UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/AtmosManager.cs

```diff
@@ -23,7 +23,8 @@ public class AtmosManager : MonoBehaviour
 
 		public HashSet<PipeData> inGameNewPipes = new HashSet<PipeData>();
 		public HashSet<FireAlarm> inGameFireAlarms = new HashSet<FireAlarm>();
-		public ConcurrentBag<PipeData> pipeToAdd = new ConcurrentBag<PipeData>();
+		private ThreadSafeList<PipeData> pipeList = new ThreadSafeList<PipeData>();
```

Comment:
isn't the problem with reagent mix being modified unsafely

InvalidOperationException: Collection was modified; enumeration operation may not execute.
System.Collections.Generic.Dictionary`2+Enumerator[TKey,TValue].MoveNext () (at <fb001e01371b4adca20013e0ac763896>:0)
Chemistry.ReagentMix.get_WholeHeatCapacity () (at /root/UnityProject/Assets/Scripts/Systems/Chemistry/ReagentMix.cs:61)
Chemistry.ReagentMix.get_InternalEnergy () (at /root/UnityProject/Assets/Scripts/Systems/Chemistry/ReagentMix.cs:79)